hk_adc/hk_adc.c

```diff
@@ -0,0 +1,530 @@
+            send_buffer = HK_ADC_REG_PTR_CONFIG;
+            break;
+
+        case HK_ADC_REG_LO_THRESH:
+            send_buffer = HK_ADC_REG_PTR_LOW_THRESH;
+            break;
+
+        case HK_ADC_REG_HI_THRESH:
+            send_buffer = HK_ADC_REG_PTR_HI_THRESH;
+            break;
+
+//        default:
+            //no other register possible
+    }
+
+    // Write to Address Pointer Register - pointing to 'reg'
+    MSS_I2C_write(&g_mss_i2c0, HK_ADC_ADDRESS_GND, &send_buffer, TX_LENGTH, MSS_I2C_RELEASE_BUS);
+
+    // Wait for completion and record the outcome
+    status = MSS_I2C_wait_complete(&g_mss_i2c0, MSS_I2C_NO_TIMEOUT);
+
+    // Proceed only if it is a success
+    if (status == MSS_I2C_SUCCESS)
+    {
+        // Read data from target slave (we always read 2B)
+        MSS_I2C_read(&g_mss_i2c0, HK_ADC_ADDRESS_GND, &rx_buffer[0], RX_LENGTH, MSS_I2C_RELEASE_BUS);
+
+        status = MSS_I2C_wait_complete(&g_mss_i2c0, MSS_I2C_NO_TIMEOUT);
+
+        if(status == MSS_I2C_SUCCESS)
+        {
+            read_value = rx_buffer[0] << 8 | rx_buffer[1];
+            *read_buffer = read_value;
+            err = HK_ADC_NO_ERR;
+            return err;
+        }
+    }
+
+    err = HK_ADC_ERR_READ_FAILED;
+    return err;
+}
+
+
+
+
+/**
+ * @brief HK ADC average voltage calculation function
+ *
+ * This local function calculates the average battery voltage from raw ADC voltage
+ * values. Number of samples to be averaged is defined by NUM_SAMPLES_TO_AVG.
+ *
+ * @param None.
+ *
+ * @return Average battery voltage raw value.
+ *
+ */
+uint16_t hk_adc_calc_avg_voltage(void)
+{
+    uint16_t sum = 0;
+    uint16_t average = 0;
+    for (int i = 0; i < NUM_SAMPLES_TO_AVG; i ++)
+    {
+        hk_adc_conv_read_volt(&volt_array[i]);
+        sum += volt_array[i];
+        average = (i == NUM_SAMPLES_TO_AVG - 1) ? sum / (uint16_t)NUM_SAMPLES_TO_AVG : 0;
+    }
+
+    return average;
+}
+
+
+
+
+/**
+ * @brief HK ADC average current calculation function
+ *
+ * This local function calculates the average battery current from raw ADC voltage
+ * values. Number of samples to be averaged is defined by NUM_SAMPLES_TO_AVG.
+ *
+ * @param None.
+ *
+ * @return Average battery current raw value.
+ *
+ */
+uint16_t hk_adc_calc_avg_current(void)
+{
+    uint16_t sum = 0;
+    uint16_t average = 0;
+    for (int i = 0; i < NUM_SAMPLES_TO_AVG; i ++)
+    {
+        hk_adc_conv_read_curr(&curr_array[i]);
+        sum += curr_array[i];
+        average = (i == NUM_SAMPLES_TO_AVG - 1) ? sum / (uint16_t)NUM_SAMPLES_TO_AVG : 0;
+    }
+
+    return average;
+}
+
+
+
+
+/**
+ * @brief HK ADC average CITIROC temperature calculation function
+ *
+ * This local function calculates the average CITIROC temperature from raw ADC voltage
+ * values. Number of samples to be averaged is defined by NUM_SAMPLES_TO_AVG.
+ *
+ * @param None.
+ *
+ * @return Average CITIROC temperature raw value.
+ *
+ */
+uint16_t hk_adc_calc_avg_citi_temp(void)
```

Comment:
Same comments as to `hk_adc_calc_avg_volt()`

Reply:
Done